CMakeLists.txt

```diff
@@ -10,7 +10,19 @@ FetchContent_Declare(
     https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz
 )
 
+# FetchContent_Declare(
+#     kokkos
+#     URL
+#     https://github.com/kokkos/kokkos/releases/download/4.5.00/kokkos-4.5.00.tar.gz
+# )
+
 FetchContent_MakeAvailable(json)
+# FetchContent_MakeAvailable(kokkos)
+
+set(CMAKE_BUILD_TYPE "Debug")
+
+set(CMAKE_CXX_FLAGS_DEBUG "-Wall -Wextra -g -O3 -march=native -fopenmp")
+set(CMAKE_CXX_FLAGS_RELEASE "-O3 -march=native -fopenmp")

```

Comment:
Same thing here. Also, adding `-O3` is redundant if CMake is compiling with `CMAKE_BUILD_TYPE=Release`, which enables it by default.